### PR TITLE
Feature/#72 add support for number values in contract definitions

### PR DESCRIPTION
**Pull request by saa14**
- Added an up script to add numbers and number formats to the ContractSchema table.
- Added methods to support numbers and formats similar to integers in ValidationExtensions and TesterService.
- Added support for numbers and number arrays in SampleData.
- Added unit tests to support number, number formats and number arrays across SampleData, Create/Edit and TesterService.

### CODE REVIEW

File: src/Totem/Features/API/SampleData.cs [hunk 1]
```diff
@@ -157,6 +162,10 @@ public static string GenerateObject(CaseInsensitiveDictionary<SchemaObject> prop
                     {
                         _jsonDictionary[key] = GenerateInteger(value.Format);
                     }
+                    if (value.Type.EqualsCaseInsensitive("number"))
```
Comment (najeraz) [inline src/Totem/Features/API/SampleData.cs]: **SHOULD**: Use `DataType` enumeration.

**Note:**
Using magic values makes the code less maintainable/refactor friendlier

File: src/Totem/Features/API/SampleData.cs [hunk 2]
```diff
@@ -202,6 +211,20 @@ public static string GenerateString(string format, string pattern)
             }
         }
 
+        public static string GenerateNumber(string format)
+        {
+            if (format != null && format.EqualsCaseInsensitive("Float"))
```
Comment (najeraz) [inline src/Totem/Features/API/SampleData.cs]: **SHOULD:** Same here use `DataType`

File: src/Totem/Services/TesterService.cs [hunk 3]
```diff
@@ -194,6 +199,32 @@ public TestMessageResult DoAllMessageValuesMatchDataTypes(CaseInsensitiveDiction
...
+            // Validate number data type
+            if (!isDouble && !isFloat)
+            {
+                AddTypeError(testMessageResult, kv.Value.ToString(), kv.Key, propertySchemaObject.Type);
+            }
+
+            if (propertySchemaObject.Format != null)
+            {
+                // Validate specific number formats
+                if (propertySchemaObject.Format.EqualsCaseInsensitive("float") && !isFloat)
```
Comment (najeraz) [inline src/Totem/Services/TesterService.cs]: **SHOULD:** Also here

Comment (saa14) [reply src/Totem/Services/TesterService.cs]: Resolved in 5ad523f0ac21a9827e4ffdaac83a3a41ec6c9ff6

File: src/Totem/Features/API/SampleData.cs (hunk 2 above)
Comment (saa14) [reply src/Totem/Features/API/SampleData.cs]: Resolved in 5ad523f

File: src/Totem.Tests/Services/TesterServiceTests.cs [hunk 4]
```diff
@@ -355,7 +400,10 @@ public void ShouldFailValidationForArrayTypeWithIncorrectItemType(string dataTyp
                     itemString = "not a datetime"; // Not a string item
                     break;
                 case "Integer":
-                    itemString = "not an integer"; // Not an integer item 
+                    itemString = "not an integer"; // Not an integer item
+                    break;
+                case "Number":
+                    itemString = "not a number"; // Not a number item
```
[the remark this answers is not in the document]
Comment (saa14) [reply src/Totem.Tests/Services/TesterServiceTests.cs]: Resolved in ae9d79523dbabe2f5c1f66b434cc8fbec78a3bf5

File: src/Totem.Tests/Services/TesterServiceTests.cs [hunk 5]
```diff
@@ -333,6 +377,7 @@ public void ShouldFailValidationForArrayTypeWithoutItems()
 
         [TestingConvention.Input("String")]
         [TestingConvention.Input("Integer")]
+        [TestingConvention.Input("Number")]
```
[the remark this answers is not in the document]
Comment (saa14) [reply src/Totem.Tests/Services/TesterServiceTests.cs]: Resolved in ae9d79523dbabe2f5c1f66b434cc8fbec78a3bf5

File: src/Totem.Tests/Features/Contracts/CreateTests.cs [hunk 6]
```diff
@@ -1076,6 +1076,47 @@ public void ShouldNotCreateWhenNonIntegerTypeExampleForIntegerType()
             command.ShouldNotValidate("The example 'not-an-integer-example' for 'Age' does not match the required data type or format 'integer'.");
         }
 
+        public void ShouldNotCreateWhenNonNumberTypeExampleForNumberType()
+        {
+            var newContract = SampleContract();
+            var command = new Create.Command()
```
[the remark this answers is not in the document]
Comment (saa14) [reply src/Totem.Tests/Features/Contracts/CreateTests.cs]: Resolved in 6eb6ebc2873bf3732a090884475c5fa38d28232b

File: src/Totem.Tests/Features/API/SampleDataTests.cs [hunk 7]
```diff
@@ -186,6 +201,23 @@ public void GeneratesProperIntegerArraySampleData()
...
+                Format = "float"
+            };
+            var sampleData = SampleData.GenerateSampleData("array", null, items: arrayItem);
+
+            var items = JsonConvert.DeserializeObject<List<string>>(sampleData);
+            foreach (var item in items)
+            {
+                var parsedItem = float.Parse(item);
+                parsedItem.GetType().ShouldBe(typeof(float));
+            }
```
[the remark this answers is not in the document]
Comment (saa14) [reply src/Totem.Tests/Features/API/SampleDataTests.cs]: Resolved in 6eb6ebc2873bf3732a090884475c5fa38d28232b

File: src/Totem.Tests/Features/API/SampleDataTests.cs [hunk 8]
```diff
@@ -97,6 +97,21 @@ public void GeneratesProperIntegerSampleData()
...
+            var floatValue = float.Parse(sampleFloat);
+            floatValue.GetType().ShouldBe(typeof(float));
+
+            var sampleDouble = SampleData.GenerateSampleData("Number", "Double");
+            var doubleValue = double.Parse(sampleDouble);
+            doubleValue.GetType().ShouldBe(typeof(double));
+
+            var sampleNumber = SampleData.GenerateSampleData("Number", null);
+            var numberValue = float.Parse(sampleNumber);
+            numberValue.GetType().ShouldBe(typeof(float));
```
[the remark this answers is not in the document]
Comment (saa14) [reply src/Totem.Tests/Features/API/SampleDataTests.cs]: Resolved in 6eb6ebc2873bf3732a090884475c5fa38d28232b

**Pull request merged**